codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/DafnyLocalServiceGenerator.java

```diff
@@ -303,8 +339,29 @@ func NewClient(clientConfig $L) (*$T, error) {
                     symbolProvider.toSymbol(outputShape)
                   )
                 );
+            returnError = "return nil,";
           }
-          returnError = "return nil,";
+        }
+        String validationCheck = "";
+        // inputShape is not a structure only when we have a positional trait
+        // TODO: Figure out how to validate constraint in a structure with positional trait
```

Comment:
My plan is to make a new PR for complete this TODO to this branch or `Golang/dev` branch